plugins/BEdita/Core/src/Model/Behavior/StatusBehavior.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2018 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\Core\Model\Behavior;
+
+use BEdita\Core\Exception\BadFilterException;
+use Cake\Core\Configure;
+use Cake\Database\Expression\QueryExpression;
+use Cake\Datasource\EntityInterface;
+use Cake\Http\Exception\BadRequestException;
+use Cake\ORM\Behavior;
+use Cake\ORM\Query;
+
+/**
+ * This behavior adds finders for object's status filtering.
+ *
+ * {@inheritDoc}
+ */
+class StatusBehavior extends Behavior
+{
+    /**
+     * Default configuration.
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'implementedMethods' => [
+            'getStatusLevel' => 'getStatusLevel',
+            'setStatusLevel' => 'setStatusLevel',
+            'checkStatus' => 'checkStatus',
+        ],
+        'implementedFinders' => [
+            'statusLevel' => 'findStatusLevel',
+        ],
+        'field' => 'status',
+    ];
+
+    /**
+     * Get the configured status level.
+     *
+     * @return string
+     */
+    public function getStatusLevel()
```

Comment:
I'm not sure if it's a good idea to have `getStatusLevel`/`setStatusLevel` methods. 
I'd prefer to keep using `Configure::read/write` explicitly throughout the code. But i's a matter of taste I guess, it is clearer for me.

Since we are changing a global configuration a call like `$Documents->setStatusLevel()` don''t simply affect the single documents table class but every object table...

Reply:
I agree.

While I don't like the `Configure::read()` approach (in my opinion it should only be used at startup, then read it through some other mean afterwards), interacting with a global setting through a behavior, which will be instantiated multiple times, is set to be more harm than good, and is likely to cause a lot of confusion.